diff -ur genres-0.9.2/Makefile genres-0.9.2-patch/Makefile
--- genres-0.9.2/Makefile	2006-04-24 22:00:20.000000000 +0200
+++ genres-0.9.2-patch/Makefile	2006-06-22 23:09:14.000000000 +0200
@@ -39,7 +39,7 @@
 #
 RELEASE=$(shell awk -F '[()]' '/\(.*\)/ { print $$2; exit; }' <debian/changelog)
 PATH:=${PATH}:/usr/local/mozilla/bin:/usr/lib/mozilla/bin:/opt/mozilla/bin
-CFLAGS= -Wall -Os -DXP_UNIX -DMOZ_X11 -DXTHREADS ` mozilla-config --cflags xpcom` -fPIC -DRELEASE="\"${RELEASE}\""
+CFLAGS= -Wall -Os -DXP_UNIX -DMOZ_X11 -DXTHREADS -I/usr/include/mozilla-firefox/plugin -I/usr/include/nspr -I/usr/include/mozilla-firefox/java -I/usr/include/mozilla-firefox/xpcom -fPIC -DRELEASE="\"${RELEASE}\""
 OBJ=main_thread.o observer_thread.o threadqueue.o slavesdir.o storage.o settings.o
 INSTALL=install -Dp -g root -o root
 LIBPREFIX=$(DESTDIR)/usr/lib
diff -ur genres-0.9.2/threadqueue.cpp genres-0.9.2-patch/threadqueue.cpp
--- genres-0.9.2/threadqueue.cpp	2006-03-22 22:02:09.000000000 +0100
+++ genres-0.9.2-patch/threadqueue.cpp	2006-06-22 23:13:32.000000000 +0200
@@ -18,7 +18,7 @@
 //    Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
 #include "nsCOMPtr.h"
-#include "nsIServiceManagerUtils.h"
+#include "nsServiceManagerUtils.h"
 #include "nsIEventQueueService.h"
 #include "nsIEventQueue.h"
 #include <stdio.h>
